cellmlmanip/_singularity_fixes.py

```diff
@@ -0,0 +1,234 @@
+"""
+`remove_fixable_singularities` specifies a method to remove fixable singularities from a model
+This represents a flattened CellML model and metadata about its variables.
+"""
+from math import isclose
+
+import networkx as nx
+from sympy import (
+    Abs,
+    Add,
+    Eq,
+    Float,
+    I,
+    Mul,
+    Piecewise,
+    Pow,
+    Wild,
+    exp,
+    log,
+    solveset,
+)
+
+import cellmlmanip
+from cellmlmanip.model import FLOAT_PRECISION
+
+
+ONE = cellmlmanip.Quantity(1.0, 'dimensionless')
+
+
+def _generate_piecewise(vs, ve, sp, ex, V):
+    """Generates a piecewise for expression based on the singularity point (sp) and vmin (vs) and vmax (ve) """
+    if vs is None:  # This shouldn't be apiecewise since we have no Vmin / Vmax
+        return ex
+
+    f_vs = ex.xreplace({V: vs})
+    f_ve = ex.xreplace({V: ve})
+    return Piecewise((f_vs + ((V - vs) / (ve - vs)) * (f_ve - f_vs),
+                      Abs(V - sp) < Abs((ve - vs) / 2)), (ex, True))
+
+
+def _get_sinularity(expr, V, U_offset, exp_function):
+    """Finds U_offset and teh singularity either side and singularity U in ghk-like equations
```

Comment:
Something like:

"Finds singularities in equations of the form::"   <-- two semicolons to start code block

Or

"Finds singularities in equations of the form:", and then "- \`\`lalala\`\`" with two \`\` (which is the syntax in restructured text. annoying, I know!)

And then below, "In addition to the singularity itself, this functions returns a lower and upper bounds such that..." and then explain whatever it is you do to determine these boundaries